torchtitan/checkpoint.py

```diff
@@ -165,7 +178,7 @@ def __init__(
         self,
         dataloader: DataLoader,
         model_parts: List[nn.Module],
-        optimizers: List[torch.optim.Optimizer],
+        optimizers: OptimizersContainer,
         lr_schedulers: List[torch.optim.lr_scheduler.LRScheduler],
```

Comment:
wouldn't the checkpointing wrapping strategies also be different for the two cases? E.g. for optimizer in bwd, the `lr_schedulers` here seems to be a **List of List**?

Reply:
changed similar to optimizers as SchedulersContainer
with the modification inside `SchedulersContainer`, SchedulersInBackwardContainer.schedulers would only output the first scheduler in each list so that it would return a **List**, same as SchedulersContainer.schedulers